app/helpers/application_helper.rb

```diff
@@ -177,7 +177,9 @@ def restful_routed_action?(controller = controller_name, action = action_name)
 
   def url_for_record(record, action = "show") # Default action is show
     @id = to_cid(record.id)
-    db  = if record.kind_of?(VmOrTemplate)
+    db  = if controller.class == VmOrTemplateController
```

Comment:
`controller.kind_of?(VmOrTemplateController)` .. but are you sure the record is never anything else in that controller? Maybe it should be `controller.kind_of?(VmOrTemplateController) && record.kind_of?(VmOrTemplate)`.

Otherwise, LGTM, seems like we really have no other way to force the quadicon to use the right controller in this case.